app.py

```diff
@@ -0,0 +1,213 @@
+    cluster=ecs_stack.cluster,
+    props=mongo_props,
+)
+
+# It is probably not appropriate host this container in ECS
+# data_props = ServiceProps(
+#     container_name="agora-data",
+#     container_location=f"ghcr.io/sage-bionetworks/agora-data:{agora_version}",
+#     container_port=9999,  # Not used
+#     container_memory=2048,
+# )
+# data_stack = ServiceStack(
+#     scope=cdk_app,
+#     construct_id=f"{stack_name_prefix}-data",
+#     vpc=network_stack.vpc,
+#     cluster=ecs_stack.cluster,
+#     props=data_props,
+#     container_env_vars={
+#         "DB_USER": "root",
+#         "DB_PASS": "changeme",
+#         "DB_NAME": "agora",
+#         "DB_PORT": "27017",
+#         "DB_HOST": "agora-mongo",
+#         "DATA_FILE": "syn13363290",
+#         "DATA_VERSION": "68",
+#         "TEAM_IMAGES_ID": "syn12861877",
+#         "SYNAPSE_AUTH_TOKEN": "agora-service-user-pat-here",
+#     },
+# )
+# data_stack.add_dependency(mongo_stack)
+
+api_props = ServiceProps(
+    container_name="agora-api",
+    container_location=f"ghcr.io/sage-bionetworks/agora-api:{agora_version}",
+    container_port=3333,
+    container_memory=1024,
+    container_env_vars={
+        "MONGODB_URI": "mongodb://root:changeme@agora-mongo:27017/agora?authSource=admin",
+        "NODE_ENV": "development",
+    },
+)
+api_stack = ServiceStack(
+    scope=cdk_app,
+    construct_id=f"{stack_name_prefix}-api",
+    vpc=network_stack.vpc,
+    cluster=ecs_stack.cluster,
+    props=api_props,
+)
+api_stack.add_dependency(mongo_stack)
+
+app_props = ServiceProps(
+    container_name="agora-app",
+    container_location=f"ghcr.io/sage-bionetworks/agora-app:{agora_version}",
+    container_port=4200,
+    container_memory=200,
+    container_env_vars={
+        "API_DOCS_URL": f"http://{fully_qualified_domain_name}/api-docs",
+        "APP_VERSION": f"{agora_version}",
+        "CSR_API_URL": f"http://{fully_qualified_domain_name}/api/v1",
+        "SSR_API_URL": "http://agora-api:3333/v1",
+    },
+)
+app_stack = ServiceStack(
+    scope=cdk_app,
+    construct_id=f"{stack_name_prefix}-app",
+    vpc=network_stack.vpc,
+    cluster=ecs_stack.cluster,
+    props=app_props,
+)
+app_stack.add_dependency(api_stack)
+
+apex_props = ServiceProps(
+    container_name="agora-apex",
+    container_location=f"ghcr.io/sage-bionetworks/agora-apex:{agora_version}",
+    container_port=80,
+    container_memory=200,
+    container_env_vars={
+        "API_DOCS_HOST": "agora-api-docs",
+        "API_DOCS_PORT": "8010",
+        "API_HOST": "agora-api",
+        "API_PORT": "3333",
+        "APP_HOST": "agora-app",
+        "APP_PORT": "4200",
+    },
+)
+apex_stack = LoadBalancedServiceStack(
+    scope=cdk_app,
+    construct_id=f"{stack_name_prefix}-apex",
+    vpc=network_stack.vpc,
+    cluster=ecs_stack.cluster,
+    props=apex_props,
+    load_balancer=load_balancer_stack.alb,
+    certificate_arn=environment_variables["CERTIFICATE_ARN"],
+    health_check_path="/health",
+)
```

Comment:
Deployment of this container is failing.   I think the problem is that the [Caddyfile in sage-monorepo](https://github.com/Sage-Bionetworks/sage-monorepo/blob/main/apps/agora/apex/Caddyfile) is not baked into the container.  The Caddyfile inside of the container does not match the one that's committed to the sage-monorepo.  Here's the Caddyfile that I see in the container..

```
# cat /etc/caddy/Caddyfile
# The Caddyfile is an easy way to configure your Caddy web server.
#
# Unless the file starts with a global options block, the first
# uncommented line is always the address of your site.
#
# To use your own domain name (with automatic HTTPS), first make
# sure your domain's A/AAAA DNS records are properly pointed to
# this machine's public IP, then replace ":80" below with your
# domain name.

:80 {
        # Set this path to your site's directory.
        root * /usr/share/caddy

        # Enable the static file server.
        file_server

        # Another common task is to set up a reverse proxy:
        # reverse_proxy localhost:8080

        # Or serve a PHP site through php-fpm:
        # php_fastcgi localhost:9000
}

# Refer to the Caddy docs for more information:
# https://caddyserver.com/docs/caddyfile
```

There are no errors in the application logs..
```

{     "level": "info",     "ts": 1732033531.8193312,     "msg": "using config from file",     "file": "/etc/caddy/Caddyfile" }
--
{     "level": "info",     "ts": 1732033531.8209982,     "msg": "adapted config to JSON",     "adapter": "caddyfile" }
{     "level": "info",     "ts": 1732033531.8491194,     "logger": "admin",     "msg": "admin endpoint started",     "address": "localhost:2019",     "enforce_origin": false,     "origins": [         "//localhost:2019",         "//[::1]:2019",         "//127.0.0.1:2019"     ] }
{     "level": "warn",     "ts": 1732033531.8494964,     "logger": "http.auto_https",     "msg": "server is listening only on the HTTP port, so no automatic HTTPS will be applied to this server",     "server_name": "srv0",     "http_port": 80 }
{     "level": "info",     "ts": 1732033531.849901,     "logger": "http.log",     "msg": "server running",     "name": "srv0",     "protocols": [         "h1",         "h2",         "h3"     ] }
{     "level": "info",     "ts": 1732033531.8502295,     "msg": "autosaved config (load with --resume flag)",     "file": "/config/caddy/autosave.json" }
{     "level": "info",     "ts": 1732033531.8503053,     "msg": "serving initial configuration" }
{     "level": "info",     "ts": 1732033531.8504949,     "logger": "tls.cache.maintenance",     "msg": "started background certificate maintenance",     "cache": "0xc0003a9b80" }
{     "level": "info",     "ts": 1732033531.8535244,     "logger": "tls",     "msg": "cleaning storage unit",     "storage": "FileStorage:/data/caddy" }
{     "level": "info",     "ts": 1732033531.8538585,     "logger": "tls",     "msg": "finished cleaning storage units" }

```

Reply:
@zaro0508 You are right. I'll open a PR to update the Docker image of `agora-apex`.

Reply:
This work is captured here: https://sagebionetworks.jira.com/browse/ARCH-327

Reply:
@tschaffter There is some argument to be had here for not baking a config file into a docker image (Thinking of re-usability of the image).

Would it make more sense to create this config file on deployment to ECS? I'm not sure how feasible it is to "just create the file". IIRC when I was looking at possibilities: the suggestion was to set up an init sidecar to create the file for you since ECS doesn't support attaching an arbitrary file on disk for it to use on fargate.

I am up for the config being a part of the docker image as long as it isn't containing anything sensitive

Reply:
@BryanFauble The Caddyfile reads its important values from environment variables.

Reply:
https://github.com/Sage-Bionetworks/sage-monorepo/blob/main/apps/agora/apex/Caddyfile

Reply:
@zaro0508 A new apex image is available ([`edge`](https://github.com/Sage-Bionetworks/sage-monorepo/pkgs/container/agora-apex/308637274?tag=edge) tag). It's probably easier to deploy the `edge` version of the stack in the dev environment for now.

Reply:
done